website/pages/docs/job-specification/volume.mdx

```diff
@@ -64,6 +64,104 @@ the [volume_mount][volume_mount] stanza in the `task` configuration.
   - `fs_type`: file system type (ex. `"ext4"`)
   - `mount_flags`: the flags passed to `mount` (ex. `"ro,noatime"`)
 
+## Volume Interpolation
+
+Because volumes represent state, many workloads with multiple allocations will
+want to mount specific volumes to specific tasks. You can use the [HCL2]
+syntax in Nomad 1.0 for fine-grained control over how volumes are used.
+
+There are two limitations to using HCL2 interpolation for `volume` blocks:
+* The `volume` block is used to schedule workloads, so any interpolation needs
+  to be done before placement. This means that variables like
+  `NOMAD_ALLOC_INDEX` can't be used for interpolation.
+* Nomad does not yet support dynamic volume creation (see [GH-8212]), so volumes
+  must be created and registered before being used as a `volume.source`.
+
+The following job specification demonstrates how to use multiple volumes with
+multiple allocations. It uses a `dynamic` block to create a new task group for
+each of the two volumes. This job specification also shows using HCL2
+variables interpolation to expose information to the task's environment.
+
+```hcl
+variables {
+  volume_index = ["0", "1"]
+  path         = "test"
+}
+
+job "example" {
+  datacenters = ["dc1"]
+
+  dynamic "group" {
+    for_each = var.volume_index
+    labels   = ["cache-${group.value}"]
+
+    content {
+
+      volume "cache-volume" {
+        type   = "csi"
+        source = "test-volume${group.value}"
+      }
+
+      network {
+        port "db" {
+          to = 6379
+        }
+      }
+
+      task "redis" {
+        driver = "docker"
+        config {
+          image = "redis:3.2"
+          ports = ["db"]
```

Comment:
What does this add? the networking stanza above directly maps to `6379` does `ports = ["db"]` have any effect?

Reply:
I haven't caught up 100% on the recent networking changes, but this job is a modification of the `nomad job init -short` example job, so I'm just keeping that as-is for now.

Reply:
Can you check that internally? I am asking because I think (especially with the networking changes) much confusion has come up about what is needed and why; so it might make sense to go over the `nomad job init -short` example once again and clean it up :)

Reply:
Sure but that's not relevant to this PR.

Reply:
@angrycub Ah I realized my mistake. I had bridge networking enabled on the `network`-stanza and somehow assumed this would be the default for group based networking. Sorry for the confusion.